### PR TITLE
fully depreciate old mappings

### PR DESCRIPTION
**Pull request by lheagy**
- add SimPEG version number to depreciation messages
- remove Depreciate.py
- remove the `_depreciate_main_map` stub
- add test for depreciation of mappings 

image

- [ ] @rowanc1: can you confirm that we will bumpversion to 0.4.0 on dev (I have added version numbers to the depreciation messages).

### CODE REVIEW

File: SimPEG/Examples/FLOW_Richards_1D_Celia1990.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def run(plotIt=True):
     def getFields(timeStep, method):
         timeSteps = np.ones(360/timeStep)*timeStep
         prob = Richards.RichardsProblem(
-            M, mapping=E,
+            M, modelMap=E,
```
Comment (rowanc1) [inline SimPEG/Examples/FLOW_Richards_1D_Celia1990.py]: this will be completely changed in a future release.

File: SimPEG/Problem.py [hunk 2]
```diff
@@ -40,24 +37,15 @@ class BaseProblem(Props.HasModel):
...
         if 'mapping' in kwargs:
-            import warnings
-            warnings.warn(
-                'The `mapping` property has been depreciated, '
-                'please use:\n\n\n'
-                '\tfrom SimPEG import Depreciate\n'
-                '\tDepreciate.use_old_mappings()\n\n\n'
-                'To bring back old functionality.'
+            raise Exception(
+                'Depreciated (in 0.4.0): use one of {}'.format(
```
Comment (rowanc1) [inline SimPEG/Problem.py]: cool.

File: SimPEG/Problem.py [hunk 3]
```diff
@@ -77,7 +65,17 @@ def mapping(self):
 
         """
```
Comment (rowanc1) [inline SimPEG/Problem.py]: Can we remove the comment from the mapping class?

File: SimPEG/Examples/FLOW_Richards_1D_Celia1990.py (hunk 1 above)
Comment (lheagy) [inline SimPEG/Examples/FLOW_Richards_1D_Celia1990.py]: fair (just needed it to pass :) )

**Comment by rowanc1** [comment]
:+1:

**Pull request merged**